gen/gen.go

```diff
@@ -44,10 +44,11 @@ func init() {
 	}
 }
 
-func Gen(jFile string, j interface{}, t *template.Template) error {
+func Gen(exercise string, j interface{}, t *template.Template) error {
 	if dirMetadata == "" {
 		return errors.New("unable to determine current path")
 	}
+	jFile := filepath.Join("exercises", exercise, "canonical-data.json")
```

Comment:
For me personally, I would 100% do it if either of the following were true:

1. they were magic numbers; numbers without explanations/names seem much harder to understand
2. A line containing "exercises" and "canonical-data.json" appeared more than once, since I don't like repetition.

I feel less stringent about these... I guess you call these magic strings? I would definitely not oppose creating a `jsonPath(exercise string) string` function (you can name that whatever you want, I'm not insistent on that name), but I also don't mind having it inlined.

Also part of my decision is that it doesn't seem very likely to change, and it is also not very hard to understand what it is doing by reading it.